lib/puppet/type/cisco_route_map.rb

```diff
@@ -0,0 +1,1642 @@
+      set_ipv4_next_hop_peer_addr            => true,
+      set_ipv4_next_hop_redist               => true,
+      set_ipv4_next_hop_unchanged            => true,
+      set_ipv4_precedence                    => 'critical',
+      set_ipv4_prefix                        => 'abcdef',
+      set_ipv6_default_next_hop              => ['2000::1', '2000::11'],
+      set_ipv6_default_next_hop_load_share   => true,
+      set_ipv6_next_hop                      => ['2000::1', '2000::11'],
+      set_ipv6_next_hop_load_share           => true,
+      set_ipv6_next_hop_peer_addr            => true,
+      set_ipv6_next_hop_redist               => true,
+      set_ipv6_next_hop_unchanged            => true,
+      set_ipv6_precedence                    => 'network',
+      set_ipv6_prefix                        => 'wxyz',
+      set_level                              => 'level-1',
+      set_local_preference                   => 100,
+      set_metric_additive                    => false,
+      set_metric_bandwidth                   => 44,
+      set_metric_delay                       => 55,
+      set_metric_reliability                 => 66,
+      set_metric_effective_bandwidth         => 77,
+      set_metric_mtu                         => 88,
+      set_metric_type                        => 'external',
+      set_nssa_only                          => true,
+      set_origin                             => 'egp',
+      set_path_selection                     => true,
+      set_tag                                => 101,
+      set_vrf                                => 'igp',
+      set_weight                             => 222,
+    }
+  "
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these
+  # patterns. These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+) (\d+) (\S+)$/,
+      [
+        [:rmname, identity],
+        [:sequence, identity],
+        [:action, identity],
+      ],
+    ]
+    patterns
+  end
+
+  # Overwrites name method. Original method simply returns self[:name],
+  # which is no longer valid or complete.
+  # Would not have failed, but just return nothing useful.
+  def name
+    "#{self[:rmname]} #{self[:sequence]} #{self[:action]}"
+  end
+
+  newparam(:name) do
+    desc 'Name of cisco_route_map, not used, but needed for puppet'
+  end
+
+  newparam(:action, namevar: true) do
+    desc 'Action for set oprtations. Valid values are permit or deny.'
+    munge(&:to_s)
+    newvalues(:permit, :deny)
+  end # param action
+
+  newparam(:sequence, namevar: true) do
+    desc "Sequence to insert/delete existing route-map entry.
+          Valid values are integer."
+  end # param sequence
+
+  newparam(:rmname, namevar: true) do
```

Comment:
For the namevars, I would list them in order that they appear in the title pattern instead of alpha order.

```
newparam(:rname..)

newparam(:sequence..)

newparam(:action..)
```